tasks/systemd.yml

```diff
@@ -0,0 +1,22 @@
+---
+- name: Systemd - Ensure service script is present.
+  become: true
+  template:
+    src:  "haproxy.systemd.j2"
+    dest: "/lib/systemd/system/haproxy.service"
+    owner: root
+    group: root
+    mode: 0755
+  register: systemd_service_file
+  notify: restart haproxy
+
+- name: Systemd - Ensure service is linked into /etc/systemd.
+  file:
+    src: "/lib/systemd/system/haproxy.service"
+    dest: "/etc/systemd/system/multi-user.target.wants/haproxy.service"
+    state: link
+
+- name: Systemd - Ensure systemd daemon is reloaded.
+  systemd:
+    daemon_reload: yes
+  when: systemd_service_file.changed
```

Comment:
Why not a handler?

Reply:
Following a private discussion it was decided to user a register + when combo in order to keep systemd specific bits in the same file.

That ok for me.